kernel/thread.c

```diff
@@ -567,6 +575,42 @@ long myst_run_thread(uint64_t cookie, uint64_t event)
     return 0;
 }
 
+long myst_run_thread(uint64_t cookie, uint64_t event)
+{
+    long ret = 0;
+    myst_thread_t* thread;
+    size_t stack_size;
+    const size_t stack_alignment = 16;
+    const size_t process_stack_size = 65536;
+    const size_t regular_stack_size = 8192;
+    uint8_t* stack = NULL;
+
+    /* get the thread corresponding to this cookie */
+    if (!(thread = _put_cookie(cookie)))
+        ERAISE(-EINVAL);
+
+    /* the stack size is determined by the thread type (process or regular) */
+    if (myst_is_process_thread(thread))
+        stack_size = process_stack_size;
+    else
+        stack_size = regular_stack_size;
```

Comment:
Why create a temporary stack for regular threads? They are going to switch to a new one on line 561 anyway.

Reply:
I wanted to do that but there may not be enough room on the OE stack for ``__run_thread`` to execute. The OE stack is only two-pages and some of that is already used by edger8r stubs and the call frame into myst_run_thread(). Also ``_run_thread`` calls other functions as well that may use up stack space. Even if there is enough space now, we would have to keep a close eye on the evolution of the ``_run_thread`` function to guarantee that it never consumes more than the remainder of the OE heap (less than two pages). Worse, we don't know how much stack is left.